src/DataTypes/Serializations/SerializationVariantElement.cpp

```diff
@@ -188,13 +189,6 @@ void SerializationVariantElement::deserializeBinaryBulkWithMultipleStreams(
             assert_cast<ColumnLowCardinality &>(*variant_element_state->variant->assumeMutable()).nestedRemoveNullable();
     }
 
-    /// If nothing to deserialize, just insert defaults.
-    if (variant_limit == 0)
```

Comment:
The problem was here. We didn't call `nested_serialization->deserializeBinaryBulkWithMultipleStreams` when there was 0 rows to read. But it's wrong, we should call it anywhay to seek the corresponding stream to the right mark if needed.